content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+
+Aside from properly document how the files inside your package
+relate to each other there is not much else you can do.
+
+You can also use this strategy by allowing access to your module via a CDN,
+by which _Component Consumers_ can simply reference the module.
+
+#### As Component Consumers
+
+This is where this approach has a hard time providing a good Consumer experience.
+
+Consumers need to:
+
+- `import module from 'module'` for Javascript, in every file that uses the module
+- Import the CSS by using either:
+  - A bundler like Webpack:
+    - `import "module.css"` only once in your Application
+  - HTML link:
+    - `cp ./node_modules/module/module.css ./public/images/dog.jpg`: copy the CSS file to your public directory
+    - `<link type="text/css" rel="stylesheet" href="public/module.css">`: load the CSS in your page.
+- Serve images at the path the module's CSS and/or Javascript files expect them, in this case at `./images/dog.jpg`
+  - `cp ./node_modules/module/images/dog.jpg ./public/images/dog.jpg`
+  - Serve `public/images/dog.jpg` by properly configuring your Application
+
+We assume that `public` is your App's public directory from where you serve your
+static files and assets, for convenience the directory is called `public` and the
+URL path is also called `public`.
+
+Notice how to _Consume_ a simple module you need to do a lot of steps and
+while there are tools to automate the process such as [Webpack][2], [Grunt][109], [Gulp][108]
+or a package like [npm-assets][107], the overall experience is not optimal.
+
+An obvious additional problem is potential _name conflicts_,
+for example if two libraries need to serve `images/dog.jpg`.
+
+_Component Authors_ might make the module available via a CDN and that might
+remove the need to fiddle around with CSS and Images but you will need to
+access the library via a Browser Global like the good old days.
+
+### Inline
+
+#### As Component Authors
+
+The bundler setup is about the same as with **Not-inline**.
+You wont be able to use [create-react-library][105] but you can probably
+find other tools that allow you to customize them to Inline everything.
+
+The Webpack configuration is actually not that hard and a working example
+will be provided at the end of this post.
+
+The key Webpack configuration is the inclusion of [url-loader][11] to
+inline images and files right into the final Javascript bundle.
+
+If you use **CSS-in-JS** libraries such as [Emotion][5] or [Styled-components][6] then
+you will be able to use all the CSS features, plus compose them via Javascript and the
+**Theming** experience will be awesome.
+You can easily create a Run-Time default Theme and provide
+hooks for _Component Consumers_ to easily change it at Run-Time.
+
+For larger Component Libraries you will have a greater experience with writing
+and maintaining CSS. You will avoid completely some CSS exclusive problems such as:
+
+- Global Scope and rule collisions
```

Comment:
I would put a period at the end of these first 2 bullet points to make it consistent with the last one.